### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by idkjs**
This is the frontend start from the other day. I had to build it seperately with bs-platform because the latest thing is that melange is not installing right now. So I built it and copied into the branch. Also, I have not run the back and front together in the same repo for the reasons we chatted with my build environment. You can tell me what happens.

### CODE REVIEW

File: frontend/Api.re [hunk 1]
```diff
@@ -1,20 +1,63 @@
...
+let fetchCounters = () =>
   Js.Promise.(
     Fetch.fetch(countersUrl)
-    |> then_(Fetch.Response.json)
-    |> then_(json => {
-      try (Counter_bs.read_counters(json) |> resolve) {
-        | exn => reject(exn)
-      }}
+    |> then_(res => Fetch.Response.json(res))
+    |> then_(json => json |> handleCounterData |> resolve)
```
Comment (davesnx) [inline frontend/Api.re:24]: I believe this works great?

```suggestion
    |> then_(json => json |> Counter_bs.read_counters |> resolve)
```

File: frontend/UI.re [hunk 2]
```diff
@@ -6,7 +6,7 @@ let unitizeWithDefault = (v, def) => switch (v) {
   | Some(v) => unitize(v)
   | None => def
 };
-
+let listToReactArray = items => items->Belt.List.toArray->React.array;
```
Comment (davesnx) [inline frontend/UI.re:9]: This doesn't seem used

**Review by davesnx — COMMENTED** [review]
You might want to do this PR against the backend one?

Aside from that, good job!

**Comment by idkjs** [comment]
merged into backend. Closing.

**Pull request closed without merging**